make/compiler_flags

```diff
@@ -62,6 +62,23 @@ CXX_MAJOR := $(shell $(CXX) -dumpversion 2>&1 | cut -d'.' -f1)
 CXX_MINOR := $(shell $(CXX) -dumpversion 2>&1 | cut -d'.' -f2)
 
 
+ifneq (,$(findstring clang,$(CXX_TYPE)))
+	CXXFLAGS_OPTIM ?= -fvectorize -ftree-vectorize -fslp-vectorize -ftree-slp-vectorize -fno-standalone-debug -fstrict-return -ftrigraphs -fvisibility=hidden -fvisibility-inlines-hidden
+	ifeq ($(shell expr $(CXX_MAJOR) \>= 5), 1)
+		CXXFLAGS_FLTO ?= -flto=full -fwhole-program-vtables -fstrict-vtable-pointers -fforce-emit-vtables
+		AR := llvm-ar-$(CXX_MAJOR)
+	endif
+endif
+ifneq (,$(findstring mingw32-g,$(CXX_TYPE)))
+else ifneq (,$(findstring gcc,$(CXX_TYPE)))
+	CXXFLAGS_OPTIM ?= -fweb -fivopts -ftree-loop-linear -fstrict-aliasing -funroll-loops -floop-strip-mine -floop-block -floop-nest-optimize -floop-unroll-and-jam -fsplit-loops -funswitch-loops -ftree-vectorize -fvect-cost-model='unlimited' -fivopts -ftree-loop-linear 	 -fvisibility=hidden -fvisibility-inlines-hidden
+	SUNDIALS_OPTIM ?= -O3 -fweb -fivopts -ftree-loop-linear
+	ifeq ($(shell expr $(CXX_MAJOR) \>= 5), 1)
```

Comment:
`expr` is available with RTools on Windows, so this is ok Windows-wise, but I think a safer option for this logic would be to check that the CXX_MAJOR is not 4 (we dont support anything below 4 anyways), which removes the need for expr here.

We already rely on on the cut gnu tool for getting CXX_MAJOR (`$(shell $(CXX) -dumpversion 2>&1 | cut -d'.' -f1)`) so take this comment as optional.

Reply:
If it's fine I'd like to leave it because we also use one compiler flag if the gcc version is greater than 7 via

```make
ifeq ($(shell expr $(CXX_MAJOR) \>= 7), 1)
```

Tho idt it's a big deal to remove that one flag and do the `eq` and `neq` if you think this will cause issues in the future

Reply:
We can keep it. Not an issue in general. In the past we were a bit too loose with the use of gnu tools, its why I have some ptsd when I see shell :). But both Rtools 35 as 40 have expr and Rtools is our suggested “toolchain” so this is cool with me.